app/models/annotation/AnnotationService.scala

```diff
@@ -966,39 +964,56 @@ class AnnotationService @Inject()(
     }
 
   //for Explorative Annotations list
-  def compactWrites(annotation: Annotation): Fox[JsObject] = {
-    implicit val ctx: DBAccessContext = GlobalAccessContext
-    for {
-      dataSet <- datasetDAO.findOne(annotation._dataSet) ?~> "dataset.notFoundForAnnotation"
-      organization <- organizationDAO.findOne(dataSet._organization) ?~> "organization.notFound"
-      teams <- teamDAO.findSharedTeamsForAnnotation(annotation._id) ?~> s"fetching sharedTeams for annotation ${annotation._id} failed"
-      teamsJson <- Fox.serialCombined(teams)(teamService.publicWrites(_, Some(organization))) ?~> s"serializing sharedTeams for annotation ${annotation._id} failed"
-      user <- userDAO.findOne(annotation._user) ?~> s"fetching owner info for annotation ${annotation._id} failed"
-      userJson = Json.obj(
-        "id" -> user._id.toString,
-        "firstName" -> user.firstName,
-        "lastName" -> user.lastName
-      )
-    } yield {
-      Json.obj(
-        "modified" -> annotation.modified,
-        "state" -> annotation.state,
-        "id" -> annotation._id.toString,
-        "name" -> annotation.name,
-        "description" -> annotation.description,
-        "typ" -> annotation.typ,
-        "stats" -> annotation.statistics,
-        "formattedHash" -> Formatter.formatHash(annotation._id.toString),
-        "annotationLayers" -> annotation.annotationLayers,
-        "dataSetName" -> dataSet.name,
-        "organization" -> organization.name,
-        "visibility" -> annotation.visibility,
-        "tracingTime" -> annotation.tracingTime,
-        "teams" -> teamsJson,
-        "tags" -> (annotation.tags ++ Set(dataSet.name, annotation.tracingType.toString)),
-        "owner" -> userJson,
-        "othersMayEdit" -> annotation.othersMayEdit
+
+  def writeCompactInfo(annotationInfo: AnnotationCompactInfo): JsObject = {
+    val teamsJson = annotationInfo.teamNames.indices.map(
+      idx =>
+        Json.obj(
+          "id" -> annotationInfo.teamIds(idx),
+          "name" -> annotationInfo.teamNames(idx),
+          "organizationId" -> annotationInfo.teamOrganizationIds(idx)
+      ))
+
+    val annotationLayerJson = annotationInfo.tracingIds.indices.map(
+      idx =>
+        Json.obj(
+          "tracingId" -> annotationInfo.tracingIds(idx),
+          "typ" -> annotationInfo.annotationLayerTypes(idx),
+          "name" -> annotationInfo.annotationLayerNames(idx)
       )
+    )
+    val tracingType: String = {
+      val skeletonPresent = annotationInfo.annotationLayerTypes.contains(AnnotationLayerType.Skeleton.toString)
+      val volumePresent = annotationInfo.annotationLayerTypes.contains(AnnotationLayerType.Volume.toString)
+      if (skeletonPresent && volumePresent) {
+        "hybrid"
+      } else if (skeletonPresent) {
+        "skeleton"
+      } else {
+        "volume"
+      }
```

Comment:
Should be extracted to a function. Maybe this already exists somewhere?

Reply:
It exists in a different form on the Annotation object, which makes it hard to use here.